vscode-dotnet-runtime-library/src/EventStream/EventStreamEvents.ts

```diff
@@ -29,12 +31,23 @@ export class EventBasedError extends Error
     }
 }
 
-export class DotnetAcquisitionStarted extends IEvent {
+export abstract class GenericModalEvent extends IEvent
+{
+    abstract readonly mode : DotnetInstallMode;
+    abstract readonly installType : DotnetInstallType;
+}
+
+export class DotnetAcquisitionStarted extends GenericModalEvent
+{
     public readonly eventName = 'DotnetAcquisitionStarted';
     public readonly type = EventType.DotnetAcquisitionStart;
+    public readonly mode;
```

Comment:
Should this have a type?

Reply:
That's a good question. The type is inherited from the base class, but I will include the type here because that's what I've done elsewhere!